src/Domain/Donation.php

```diff
@@ -442,35 +442,11 @@ public function replaceNullPaymentMethodTypeWithCard(): void
         $this->paymentMethodType = PaymentMethodType::Card;
     }
 
-    /**
-     * @return array A json encode-ready array representation of the donation, for sending to Salesforce.
-     */
-    public function toHookModel(): array
-    {
-        $data = $this->toApiModel();
-
-        $data['updatedTime'] = $this->getUpdatedDate()->format(DateTimeInterface::ATOM);
-        $data['amountMatchedByChampionFunds'] = (float) $this->getConfirmedChampionWithdrawalTotal();
-        $data['amountMatchedByPledges'] = (float) $this->getConfirmedPledgeWithdrawalTotal();
-        $data['originalPspFee'] = (float) $this->getOriginalPspFee();
-        $data['refundedTime'] = $this->refundedAt?->format(DateTimeInterface::ATOM);
-        $data['tbgGiftAidRequestConfirmedCompleteAt'] =
-            $this->tbgGiftAidRequestConfirmedCompleteAt?->format(DateTimeInterface::ATOM);
-        unset(
-            $data['charityName'],
-            $data['donationId'],
-            $data['matchReservedAmount'],
-            $data['matchedAmount'],
-            $data['cardBrand'],
-            $data['cardCountry'],
-        );
-
-        return $data;
-    }
-
-    public function toApiModel(): array
+    public function toApiModel(bool $forSalesforce = false): array
     {
         $data = [
+            'amountMatchedByChampionFunds' => (float) $this->getConfirmedChampionWithdrawalTotal(),
```

Comment:
Seems a shame to convert to float when we've been so careful to always treat as strings up to now, and it will be converted to Decimal in salesforce. But should be fine and not much we can do. Made me find this PHP issue - would be good if it was fixed in PHP https://bugs.php.net/bug.php?id=80484